graf2d/gpad/src/TPad.cxx

```diff
@@ -1005,60 +1011,64 @@ Int_t TPad::ClipPolygon(Int_t n, Double_t *x, Double_t *y, Int_t nn, Double_t *x
       x1 = x2; y1 = y2;
    }
 
-   if (nc>0) {
-
-      // Clip against the right boundary
-      x1 = xc[nc-1]; y1 = yc[nc-1];
-      nc2 = 0;
-      for (i=0; i<nc; i++) {
-         x2 = xc[i]; y2 = yc[i];
-         if (x1 == x2) {
-            slope = 0;
+   // Clip against the right boundary
+   if (nc > 0) {
+      x1 = xc[nc - 1];
+      y1 = yc[nc - 1];
+   }
```

Comment:
You let execute now complete branch which was before checked with `if (nc > 0)`.
Is it really necessary?

Reply:
It's not executed since `for ( ; i < 0 ; )` never enters the loop. This way, we set nc2 = 0 before going to the next dimension.

Reply:
But it completely useless. 
If `nc == 0` there is nothing to do, why you want to check this several more times?

Reply:
Fair enough.